test cases/common/139 simd/meson.build

```diff
@@ -0,0 +1,31 @@
+project('simd', 'c')
+
+simd = import('simd')
+
+cc = meson.get_compiler('c')
+
+cdata = configuration_data()
+
+simdlibs = simd.check('mysimds',
+  mmx : 'simd_mmx.c',
+  sse : 'simd_sse.c',
+  sse2 : 'simd_sse2.c',
+  sse3 : 'simd_sse3.c',
+  ssse3 : 'simd_ssse3.c',
+  sse41 : 'simd_sse41.c',
+  sse42 : 'simd_sse42.c',
+  avx : 'simd_avx.c',
+  avx2 : 'simd_avx2.c',
+  neon : 'simd_neon.c',
+  configuration : cdata,
```

Comment:
I'm not sure whether I like the idea of functions modifying kwargs in-place. Nothing else does this, and we should be trying to keep the functions in our build language as 'pure' as possible. 

I think it's better to split this functionality into two parts:

1) Function that takes a cdata and returns a copy of it with values set
2) Function that takes a cdata + sources and returns a static library

Or, return an array with the cdata as the second argument, but I think the consensus is that array return values in Meson are ugly.